janis_bioinformatics/tools/common/bwaaligner.py

```diff
@@ -75,6 +84,58 @@ def bind_metadata(self):
         self.metadata.dateCreated = "2018-12-24"
         self.metadata.version = "1.1"
 
+    def tests(self):
+        return [
+            TTestCase(
+                name="basic",
+                input={
+                    "sample_name": "NA12878",
+                    "fastq": [
+                        os.path.join(
+                            BioinformaticsWorkflow.test_data_path(), "BRCA1_R1.fastq.gz"
+                        ),
+                        os.path.join(
+                            BioinformaticsWorkflow.test_data_path(), "BRCA1_R2.fastq.gz"
+                        ),
+                    ],
+                    "reference": os.path.join(
+                        BioinformaticsWorkflow.test_data_path(), "hg38-brca1.fasta"
+                    ),
+                },
+                output=[
+                    TTestExpectedOutput(
+                        tag="out",
+                        preprocessor=TTestPreprocessor.FileSize,
+                        operator=operator.gt,
+                        expected_value=2767780,
+                    ),
+                    TTestExpectedOutput(
+                        tag="out",
+                        suffix_secondary_file=".bai",
+                        preprocessor=TTestPreprocessor.FileSize,
+                        operator=operator.gt,
+                        expected_value=290,
+                    ),
+                    TTestExpectedOutput(
+                        tag="out",
+                        preprocessor=Bam.flagstat,
+                        operator=operator.eq,
+                        expected_file=os.path.join(
+                            self.test_data_path(), "bwaaligner.flagstat.txt"
+                        ),
+                    ),
+                    TTestExpectedOutput(
+                        tag="out",
+                        preprocessor=TTestPreprocessor.Value,
+                        operator=Bam.equal,
+                        expected_value=os.path.join(
+                            self.test_data_path(), "bwaaligner.bam"
+                        ),
```

Comment:
This is an example of 2 different ways we can check if 2 bam files are the same using custom preprocessor or custom operator.

Reply:
This is fantastic, @rlupat FYI.